tsdb/raw.go

```diff
@@ -93,6 +93,7 @@ func (e *RawExecutor) execute(out chan *models.Row) {
 	// Keep looping until all mappers drained.
 	var err error
 	for {
+		timeout := time.After(30 * time.Second)
```

Comment:
If we move the After() call to the select then it's only the read time and not the processing time.